x-pack/plugins/security_solution/public/resolver/store/ui/selectors.ts

```diff
@@ -31,20 +35,129 @@ export const selectedNode = createSelector(
 );
 
 /**
- * The legacy `crumbEvent` and `crumbId` parameters.
- * @deprecated
+ * Which view should show in the panel, as well as what parameters should be used.
+ * Calculated using the query string
  */
-export const breadcrumbParameters = createSelector(
+export const panelViewAndParameters = createSelector(
   (state: ResolverUIState) => state.locationSearch,
   (state: ResolverUIState) => state.resolverComponentInstanceID,
   (locationSearch, resolverComponentInstanceID) => {
     if (locationSearch === undefined || resolverComponentInstanceID === undefined) {
       // Equivalent to `null`
-      return {
-        crumbId: '',
-        crumbEvent: '',
-      };
+      return defaultParameters();
+    }
+    const urlSearchParams = new URLSearchParams(locationSearch);
+    const value = urlSearchParams.get(parameterName(resolverComponentInstanceID));
+    if (value === null) {
+      // Equivalent to `null`
+      return defaultParameters();
+    }
+    const decodedValue: unknown = decode(value);
+    if (isPanelViewAndParameters(decodedValue)) {
+      return decodedValue;
     }
-    return locationSearchModel.breadcrumbParameters(locationSearch, resolverComponentInstanceID);
+    return defaultParameters();
   }
 );
+
+/**
+ * Return a relative href (which includes just the 'search' part) that contains an encoded version of `params `.
+ * All other values in the 'search' will be kept.
+ * Use this to get an `href` for an anchor tag.
+ */
+export const relativeHref: (
+  state: ResolverUIState
+) => (params: PanelViewAndParameters) => string | undefined = createSelector(
+  (state: ResolverUIState) => state.locationSearch,
+  (state: ResolverUIState) => state.resolverComponentInstanceID,
+  (locationSearch, resolverComponentInstanceID) => {
+    return (params: PanelViewAndParameters) => {
+      /**
+       * This is only possible before the first `'appReceivedNewExternalProperties'` action is fired.
+       */
+      if (locationSearch === undefined || resolverComponentInstanceID === undefined) {
+        return undefined;
+      }
+      const urlSearchParams = new URLSearchParams(locationSearch);
+      const value = encode(params);
+      urlSearchParams.set(parameterName(resolverComponentInstanceID), value);
+      return `?${urlSearchParams.toString()}`;
+    };
+  }
+);
+
+/**
+ * Returns a map of ecs category name to urls for use in panel navigation.
+ * @deprecated
+ */
+export const relatedEventsRelativeHrefs: (
+  state: ResolverUIState
+) => (
+  categories: Record<string, number> | undefined,
+  nodeID: string
+) => Map<string, string | undefined> = createSelector(relativeHref, (relativeHref) => {
+  return (categories: Record<string, number> | undefined, nodeID: string) => {
+    const hrefsByCategory = new Map<string, string | undefined>();
+    if (categories !== undefined) {
+      Object.keys(categories).map((category) => {
```

Comment:
Should this be a `for (const category of Object.keys(cateogires)) {` ? I might be misunderstanding the the `map` call but we're just iterating over the array and adding the `categoryPanelParams` to a new `Map` right?

What's the benefit of using the `map` function here?

Reply:
yes it should

Reply:
Matter of taste. Both iterate over a collection... technically using a `.map` lures the reader here into thinking this is more purely functional when it isn't, but I think that's OK.

Reply:
fwiw they are effectively the same thing, mostly just a habit from before `for (... of ...)` was even a thing